### PR TITLE
JAMES-3494 Update reactor version

### PR DESCRIPTION
**Pull request by remk**
Update reactor in order to get have the last version of reactor netty which will enable us to use websockets.

This PR does not aim to fix the now deprecated Elastic schedulers.

### CODE REVIEW

File: server/protocols/jmap-rfc-8621/src/main/scala/org/apache/james/jmap/method/EmailSetUpdatePerformer.scala [hunk 1]
```diff
@@ -169,22 +169,19 @@ class EmailSetUpdatePerformer @Inject() (serializer: EmailSetSerializer,
...
+                                  .keys
+                                  .toSeq
+                                SMono.fromCallable[Seq[EmailUpdateResult]](() => {
+                                  operation.accept(range)
+                                  messageIds.map(EmailUpdateSuccess)
+                                })
+                                  .onErrorResume(e => SMono.just(messageIds.map(id => EmailUpdateFailure(EmailSet.asUnparsed(id), e))))
+                                  .subscribeOn(Schedulers.elastic())
+                              })
+                              .reduce(Seq[EmailUpdateResult]())( _ ++ _)
```
Comment (chibenwa) [inline server/protocols/jmap-rfc-8621/src/main/scala/org/apache/james/jmap/method/EmailSetUpdatePerformer.scala:184]: Can we avoid this reindent?

**Review by chibenwa — APPROVED** [review]

**Comment by chibenwa** [comment]
```

[c06e911241afe1828a6389d3a51f2276a4f2175e] [ERROR]   MemoryTaskManagerTest.awaitShouldAwaitWaitingTask » Arithmetic long overflow
[c06e911241afe1828a6389d3a51f2276a4f2175e] [ERROR]   MemoryTaskManagerTest.awaitShouldNotThrowWhenCompletedTask » Arithmetic long o...
[c06e911241afe1828a6389d3a51f2276a4f2175e] [ERROR]   EventSourcingTaskManagerTest.awaitShouldAwaitWaitingTask » Arithmetic long ove...
[c06e911241afe1828a6389d3a51f2276a4f2175e] [ERROR]   EventSourcingTaskManagerTest.awaitShouldNotThrowWhenCompletedTask » Arithmetic
```

```
[c06e911241afe1828a6389d3a51f2276a4f2175e] [ERROR]   EventSourcingTaskManagerTest.awaitANonExistingTaskShouldReturnAnUnknownAwaitedTaskExecutionDetailsAndThrow 
[c06e911241afe1828a6389d3a51f2276a4f2175e] Expecting:
[c06e911241afe1828a6389d3a51f2276a4f2175e]   <java.lang.ArithmeticException: long overflow>
[c06e911241afe1828a6389d3a51f2276a4f2175e] to be an instance of:
[c06e911241afe1828a6389d3a51f2276a4f2175e]   <org.apache.james.task.TaskNotFoundException>
[c06e911241afe1828a6389d3a51f2276a4f2175e] but was:
[c06e911241afe1828a6389d3a51f2276a4f2175e]   <"java.lang.ArithmeticException: long overflow
[c06e911241afe1828a6389d3a51f2276a4f2175e] 	at java.base/java.lang.Math.multiplyExact(Math.java:946)
[c06e911241afe1828a6389d3a51f2276a4f2175e] 	at java.base/java.time.Duration.toNanos(Duration.java:1248)
[c06e911241afe1828a6389d3a51f2276a4f2175e] 	at reactor.core.publisher.Flux.blockFirst(Flux.java:2488)
[c06e911241afe1828a6389d3a51f2276a4f2175e] 	at org.apache.james.task.eventsourcing.EventSourcingTaskManager.await(EventSourcingTaskManager.scala:119)
```

**Comment by Arsnael** [comment]
```
[181b7dfa74cd69965305782f5dae8f60a351b785] [ERROR] reIndexMailboxShouldWaitsForExecution  Time elapsed: 1.147 s  <<< ERROR!
[181b7dfa74cd69965305782f5dae8f60a351b785] java.lang.ArithmeticException: long overflow
[181b7dfa74cd69965305782f5dae8f60a351b785] 	at org.apache.james.adapter.mailbox.ReIndexerManagementTest.reIndexMailboxShouldWaitsForExecution(ReIndexerManagementTest.java:62)
[181b7dfa74cd69965305782f5dae8f60a351b785] 
[181b7dfa74cd69965305782f5dae8f60a351b785] [ERROR] reIndexShouldWaitsForExecution  Time elapsed: 0.007 s  <<< ERROR!
[181b7dfa74cd69965305782f5dae8f60a351b785] java.lang.ArithmeticException: long overflow
[181b7dfa74cd69965305782f5dae8f60a351b785] 	at org.apache.james.adapter.mailbox.ReIndexerManagementTest.reIndexShouldWaitsForExecution(ReIndexerManagementTest.java:74)
```

**Review by Arsnael — COMMENTED** [review]
I read it and have nothing to say about the code, but regarding the build results it looks rather unstable...

**Comment by chibenwa** [comment]
> I read it and have nothing to say about the code, but regarding the build results it looks rather unstable...

Pretty much expected: it will keep breaking where we await the task manager with no defined duration.

The build here is not `unstable` but `broken`. Nothing to be worried of!

**Comment by chibenwa** [comment]
`[INFO] Apache James :: Server :: Cassandra - guice injection FAILURE [20:05 min]`

[INFO] Running org.apache.james.CassandraJamesServerTest

[ERROR] Failed to execute goal org.apache.maven.plugins:maven-surefire-plugin:2.22.2:test (default-test) on project james-server-cassandra-guice: There was a timeout or other error in the fork -> [Help 1]

test this please

**Comment by Arsnael** [comment]
```
[a62f7424a42cd43690c1646bb100595b2c366360] [ERROR] Errors: 
[a62f7424a42cd43690c1646bb100595b2c366360] [ERROR]   DSNRelayTest.setUp:119 » Retryable Unexpected end of file from server executin...
```

**Comment by chibenwa** [comment]
test this please

**Comment by Arsnael** [comment]
Quite a big change here... No ticket?

**Comment by chibenwa** [comment]
`[ERROR]   CassandraSetMessagesMethodTest>SetMessagesMethodTest.sendingAMailShouldLeadToAppropriateMailboxCountersOnSent:1248 1 expectation failed.`

Unstable?

test this please

**Review by chibenwa — APPROVED** [review]

**Comment by chibenwa** [comment]
Woooohooooo :green_apple: koodo Remi!!!

**Review by Arsnael — APPROVED** [review]

**Comment by Arsnael** [comment]
Merged

**Pull request closed without merging**